zppy/templates/pcmdi_diags.bash

```diff
@@ -0,0 +1,1308 @@
+        f"but got {len(ts_periods)}: {ts_periods}"
+    )
+
+# Set up paths
+obs_dir = os.path.join('{{pcmdi_external_prefix}}', 'observations', 'Atm', 'time-series')
+pmp_dir = os.path.join('{{pcmdi_external_prefix}}', 'pcmdi_data')
+out_dir = os.path.join("${web_dir}", "${results_dir}", "viewer")
+os.makedirs(out_dir, exist_ok=True)
+
+# Copy logo
+web_logo_src = os.path.join(
+    '{{pcmdi_external_prefix}}',
+    '{{pcmdi_viewer_template}}',
+    'e3sm_pmp_logo.png'
+)
+web_logo_dst = os.path.join(out_dir, 'e3sm_pmp_logo.png')
+shutil.copy(web_logo_src, web_logo_dst)
+
+# Build config
+config = collect_config(
+    title=title,
+    subtitle=subtitle,
+    version=version,
+    case_id=case_id,
+    diag_dir="${web_dir}",
+    obs_dir=obs_dir,
+    pmp_dir=pmp_dir,
+    out_dir=out_dir,
+    clim_period=clim_period,
+    emov_period=emov_period,
+    enso_period=enso_period
+)
+
+# Render viewer
+generate_methodology_html(config)
+generate_data_html(config)
+generate_viewer_html(config)
+
+{% endif %}
+
+EOF
+################################
+# Run diagnostics
+mkdir -p pcmdi_diags
+command="srun -N 1 python -u pcmdi.py"
+# Run diagnostics
+time ${command}
+if [ $? != 0 ]; then
+  cd {{ scriptDir }}
+  echo 'ERROR (11)' > {{ prefix }}.status
+  exit 11
+fi
+
+#################################
+# Copy output to web server
+echo
+echo ===== COPY FILES TO WEB SERVER =====
+echo
+
+# Create top-level directory
+mkdir -p ${web_dir}
+if [ $? != 0 ]; then
+  cd {{ scriptDir }}
+  echo 'ERROR (13)' > {{ prefix }}.status
+  exit 13
+fi
+
+{% if machine in ['pm-cpu', 'pm-gpu'] %}
+# For NERSC, make sure it is world readable
+f=`realpath ${web_dir}`
+while [[ $f != "/" ]]
+do
+  owner=`stat --format '%U' $f`
+  if [ "${owner}" = "${USER}" ]; then
+    chgrp e3sm $f
+    chmod go+rx $f
+  fi
+  f=$(dirname $f)
+done
+{% endif %}
+
+############################################
+# Copy files
+#rsync -a --delete ${results_dir} ${web_dir}/
+{% if "synthetic_plots" not in subsection %}
+rsync -a ${results_dir} ${web_dir}/
+if [ $? != 0 ]; then
+  cd {{ scriptDir }}
+  echo 'ERROR (14)' > {{ prefix }}.status
+  exit 14
+fi
+{% endif %}
+
+{% if machine in ['pm-cpu', 'pm-gpu'] %}
+# For NERSC, change permissions of new files
+pushd ${web_dir}/
+chgrp -R e3sm ${results_dir}
+chmod -R go+rX,go-w ${results_dir}
+popd
+{% endif %}
+
+{% if machine in ['anvil', 'chrysalis'] %}
+# For LCRC, change permissions of new files
+pushd ${web_dir}/
+chmod -R go+rX,go-w ${results_dir}
+popd
+{% endif %}
+
+# Delete temporary workdir
+cd ..
+if [[ "${debug,,}" != "true" ]]; then
+  rm -rf ${workdir}
+fi
+
+# Update status file and exit
+{% raw %}
+ENDTIME=$(date +%s)
+ELAPSEDTIME=$(($ENDTIME - $STARTTIME))
+{% endraw %}
+echo ==============================================
+echo "Elapsed time: $ELAPSEDTIME seconds"
+echo ==============================================
+rm -f {{ prefix }}.status
+echo 'OK' > {{ prefix }}.status
+exit 0
```

Comment:
We should move some of this into `zppy-interfaces`. We shouldn't have a 1300-line bash file in `zppy`; these bash files should be pretty minimal.